### PR TITLE
feat: Parameter Add: OIDC Database Users

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR adds support for OIDC federated authentication user
Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments

FYI: We cannot define an acceptance test because we need to set a federation authentication sets in the Org
### Testing 
I tested my changes locally by building the provider binary with my changes. 

#### TF Configuration
```terraform
resource "mongodbatlas_database_user" "test" {
  username           = "64d613677e1ad50839cce4db/testAndrea22"
  project_id         = "6414908c207f4d22f4d8f232"
  auth_database_name = "admin"
  oidc_auth_type     = "IDP_GROUP"

  roles {
    role_name     = "readWriteAnyDatabase"
    database_name = "admin"
  }
}

data "mongodbatlas_database_user" "test" {
  username           = mongodbatlas_database_user.test.username
  project_id         = "6414908c207f4d22f4d8f232"
  auth_database_name = "admin"
}

data "mongodbatlas_database_users" "test" {
  project_id = "6414908c207f4d22f4d8f232"
}
```

####  terraform plan|apply|destroy
```bash
Terraform will perform the following actions:

  # mongodbatlas_database_user.test will be created
  + resource "mongodbatlas_database_user" "test" {
      + auth_database_name = "admin"
      + aws_iam_type       = "NONE"
      + id                 = (known after apply)
      + ldap_auth_type     = "NONE"
      + oidc_auth_type     = "IDP_GROUP"
      + project_id         = "6414908c207f4d22f4d8f232"
      + username           = "64d613677e1ad50839cce4db/testAndrea22"
      + x509_type          = "NONE"

      + roles {
          + collection_name = (known after apply)
          + database_name   = "admin"
          + role_name       = "readWriteAnyDatabase"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.



{
 "awsIAMType": "NONE",
 "databaseName": "admin",
 "groupId": "6414908c207f4d22f4d8f232",
 "labels": [],
 "ldapAuthType": "NONE",
 "links": [
  {
   "href": "https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/6414908c207f4d22f4d8f232/databaseUsers/admin/64d613677e1ad50839cce4db%2FtestAndrea22",
   "rel": "self"
  }
 ],
 "oidcAuthType": "IDP_GROUP",
 "roles": [
  {
   "databaseName": "admin",
   "roleName": "readWriteAnyDatabase"
  }
 ],
 "scopes": [],
 "username": "64d613677e1ad50839cce4db/testAndrea22",
 "x509Type": "NONE"
}
-----------------------------------------------------: timestamp=2023-08-11T12:21:22.725+0100
mongodbatlas_database_user.test: Creation complete after 1s [id=YXV0aF9kYXRhYmFzZV9uYW1l:YWRtaW4=-cHJvamVjdF9pZA==:NjQxNDkwOGMyMDdmNGQyMmY0ZDhmMjMy-dXNlcm5hbWU=:NjRkNjEzNjc3ZTFhZDUwODM5Y2NlNGRiL3Rlc3RBbmRyZWEyMg==]
2023-08-11T12:21:22.747+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-08-11T12:21:22.749+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-11T12:21:22.751+0100 [DEBUG] provider: plugin process exited: path=../bin/terraform-provider-mongodbatlas pid=89817
2023-08-11T12:21:22.751+0100 [DEBUG] provider: plugin exited

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.



mongodbatlas_database_user.test: Destruction complete after 1s
2023-08-11T12:21:57.643+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-08-11T12:21:57.647+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-11T12:21:57.656+0100 [DEBUG] provider: plugin process exited: path=../bin/terraform-provider-mongodbatlas pid=90102
2023-08-11T12:21:57.656+0100 [DEBUG] provider: plugin exited

Destroy complete! Resources: 1 destroyed.
```

#### Terraform state
```bash
terraform state show data.mongodbatlas_database_user.test
# data.mongodbatlas_database_user.test:
data "mongodbatlas_database_user" "test" {
    auth_database_name = "admin"
    aws_iam_type       = "NONE"
    id                 = "YXV0aF9kYXRhYmFzZV9uYW1l:YWRtaW4=-cHJvamVjdF9pZA==:NjQxNDkwOGMyMDdmNGQyMmY0ZDhmMjMy-dXNlcm5hbWU=:NjRkNjEzNjc3ZTFhZDUwODM5Y2NlNGRiL3Rlc3RBbmRyZWEyMg=="
    labels             = []
    ldap_auth_type     = "NONE"
    oidc_auth_type     = "IDP_GROUP"
    project_id         = "6414908c207f4d22f4d8f232"
    roles              = [
        {
            collection_name = ""
            database_name   = "admin"
            role_name       = "readWriteAnyDatabase"
        },
    ]
    scopes             = []
    username           = "64d613677e1ad50839cce4db/testAndrea22"
    x509_type          = "NONE"



terraform state show data.mongodbatlas_database_users.test
# data.mongodbatlas_database_users.test:
data "mongodbatlas_database_users" "test" {
    id         = "terraform-20230811113157412700000001"
    project_id = "6414908c207f4d22f4d8f232"
    results    = [
        {
            auth_database_name = "admin"
            aws_iam_type       = "NONE"
            labels             = []
            ldap_auth_type     = "NONE"
            oidc_auth_type     = "IDP_GROUP"
            project_id         = "6414908c207f4d22f4d8f232"
            roles              = [
                {
                    collection_name = ""
                    database_name   = "admin"
                    role_name       = "readWriteAnyDatabase"
                },
            ]
            scopes             = []
            username           = "64d613677e1ad50839cce4db/testAndrea"
            x509_type          = "NONE"
        },
        {
            auth_database_name = "admin"
            aws_iam_type       = "NONE"
            labels             = []
            ldap_auth_type     = "NONE"
            oidc_auth_type     = "IDP_GROUP"
            project_id         = "6414908c207f4d22f4d8f232"
            roles              = [
                {
                    collection_name = ""
                    database_name   = "admin"
                    role_name       = "readWriteAnyDatabase"
                },
            ]
            scopes             = []
            username           = "64d613677e1ad50839cce4db/testAndrea22"
            x509_type          = "NONE"
        },
    ]
}
```
Screenshot 2023-08-11 at 12 21 50

### CODE REVIEW

File: website/docs/r/database_user.html.markdown [hunk 1]
```diff
@@ -117,7 +135,7 @@ Accepted values include:
   * `MANAGED` - The user is being created for use with Atlas-managed X.509.Externally authenticated users can only be created on the `$external` database.
   * `CUSTOMER` -  The user is being created for use with Self-Managed X.509. Users created with this x509Type require a Common Name (CN) in the username field. Externally authenticated users can only be created on the `$external` database.
 
-* `aws_iam_type` - (Optional) If this value is set, the new database user authenticates with AWS IAM credentials. If no value is given, Atlas uses the default value of NONE. The accepted types are:
+* `aws_iam_type` - (Optional) If this value is set, the new database user authenticates with AWS IAM credentials. If no value is given, Atlas uses the default value of `NONE``. The accepted types are:
```
Comment (gssbzn) [inline website/docs/r/database_user.html.markdown]:
```suggestion
* `aws_iam_type` - (Optional) If this value is set, the new database user authenticates with AWS IAM credentials. If no value is given, Atlas uses the default value of `NONE`. The accepted types are:
```

**Review by gssbzn — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]

File: website/docs/d/database_user.html.markdown [hunk 2]
```diff
@@ -64,6 +64,9 @@ In addition to all arguments above, the following attributes are exported:
 * `x509_type` - X.509 method by which the provided username is authenticated.
 * `aws_iam_type` - The new database user authenticates with AWS IAM credentials. Default is `NONE`, `USER` means user has AWS IAM user credentials, `ROLE` - means user has credentials associated with an AWS IAM role.
 * `ldap_auth_type` - Method by which the provided username is authenticated. Default is `NONE`. Other valid values are: `USER`, `GROUP`.
+* `oidc_auth_type` - (Optional) Human-readable label that indicates whether the new database user authenticates with OIDC federated authentication. If no value is given, Atlas uses the default value of `NONE`. The accepted types are:
```
Comment (Zuhairahmed) [inline website/docs/d/database_user.html.markdown]:
```suggestion
* `oidc_auth_type` - (Optional) Human-readable label that indicates whether the new database user authenticates with OIDC (OpenID Connect) federated authentication. If no value is given, Atlas uses the default value of `NONE`. The accepted types are:
```

File: website/docs/d/database_users.html.markdown [hunk 3]
```diff
@@ -76,6 +76,9 @@ Possible values include:
 * `x509_type` - X.509 method by which the provided username is authenticated.
 * `aws_iam_type` - The new database user authenticates with AWS IAM credentials. Default is `NONE`, `USER` means user has AWS IAM user credentials, `ROLE` - means user has credentials associated with an AWS IAM role.
 * `ldap_auth_type` - Method by which the provided username is authenticated. Default is `NONE`. Other valid values are: `USER`, `GROUP`.
+* `oidc_auth_type` - (Optional) Human-readable label that indicates whether the new database user authenticates with OIDC federated authentication. If no value is given, Atlas uses the default value of `NONE`. The accepted types are:
```
Comment (Zuhairahmed) [inline website/docs/d/database_users.html.markdown]:
```suggestion
* `oidc_auth_type` - (Optional) Human-readable label that indicates whether the new database user authenticates with OIDC (OpenID Connect) federated authentication. If no value is given, Atlas uses the default value of `NONE`. The accepted types are:
```

File: website/docs/r/database_user.html.markdown [hunk 4]
```diff
@@ -127,6 +145,10 @@ Accepted values include:
   * `USER` - LDAP server authenticates this user through the user's LDAP user. `username` must also be a fully qualified distinguished name, as defined in [RFC-2253](https://tools.ietf.org/html/rfc2253).
   * `GROUP` - LDAP server authenticates this user using their LDAP user and authorizes this user using their LDAP group. To learn more about LDAP security, see [Set up User Authentication and Authorization with LDAP](https://docs.atlas.mongodb.com/security-ldaps). `username` must also be a fully qualified distinguished name, as defined in [RFC-2253](https://tools.ietf.org/html/rfc2253).
 
+
+* `oidc_auth_type` - (Optional) Human-readable label that indicates whether the new database user authenticates with OIDC federated authentication. If no value is given, Atlas uses the default value of `NONE`. The accepted types are:
```
Comment (Zuhairahmed) [inline website/docs/r/database_user.html.markdown]:
```suggestion
* `oidc_auth_type` - (Optional) Human-readable label that indicates whether the new database user authenticates with OIDC (OpenID Connect) federated authentication. If no value is given, Atlas uses the default value of `NONE`. The accepted types are:
```

**Review by Zuhairahmed — APPROVED** [review]
LGTM, minor comments. Nice! Customers are going to love this one

**Review by zach-carr — COMMENTED** [review]
Retroactively commenting - this looks good copy-wise 👍

**Pull request merged**